### PR TITLE
2797 Add Table of Contents Skip Link

### LINKED ISSUE #2797: Add "Skip Link" to skip over Jump to Section
To improve keyboard experience, we should add a skip link before the "Jump to Section" to allow keyboard navigators to skip over the jump to section.

### PR DESCRIPTION
**Pull request by JCoder121**
[Deploy Preview](https://deploy-preview-2808--keen-mayer-a86c8b.netlify.app/components/card)

#### What does this PR do?
Add additional SkipLink component for pages that have a Table of Contents, to allow the user to navigate past the Table of Contents using the keyboard.

#### Where should the reviewer start?
aries-site/src/layouts/main/Layout.js

#### What testing has been done on this PR?
Manual visual testing on local build.

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [x] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [x] Visual snapshots are reasonable

#### How should this be manually tested?
Ensure that SkipLink for Table of Contents is present only when necessary and that all previous functionality is not broken.

#### Any background context you want to provide?
Previously, using the keyboard to navigate through the DS page would require a user to manually tab through every single header section in the Table of Contents. By providing a SkipLink at the beginning of the page, the user has the choice to skip keyboard navigation of the ToC.

#### What are the relevant issues?
Closes #2797.  

#### Screenshots (if appropriate)
On pages with Table of Contents:
image

On pages without Table of Contents:
image

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: aries-site/src/layouts/main/Layout.js [hunk 1]
```diff
@@ -63,6 +63,14 @@ export const Layout = ({
   const showInPageNav =
     !['xsmall', 'small'].includes(size) && headings?.length > 0;
 
+  /* If no headings are found, do not show Table of Contents SkipLink, 
+     instead set ToC skiplink as undefined and filter it out.
+     Future: Enhance Grommet SkipLink component to automatically filter */
```
Comment (taysea) [inline aries-site/src/layouts/main/Layout.js:68]:
```suggestion
     Logic to be removed in future by: https://github.com/grommet/grommet/issues/6266  */
```

File: aries-site/src/layouts/main/Layout.js [hunk 2]
```diff
@@ -84,20 +92,22 @@ export const Layout = ({
           />
           <>
             <SkipLinks id="skip-links">
-              <SkipLink id="main" label="Main Content" />
+              {skiplinks.map(({ id, label }) => (
+                <SkipLink id={id} label={label} />
```
Comment (taysea) [inline aries-site/src/layouts/main/Layout.js:96]:
```suggestion
                <SkipLink key={id} id={id} label={label} />
```

Comment (taysea) [reply aries-site/src/layouts/main/Layout.js:96]: Mapping requires each item have a key passed onto it.

**Review by taysea — APPROVED** [review]
LGTM.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**